### ISSUE TITLE
RemoveIsolatedNodes(): both dimendions of train_mask and data.y still contain the removed isolated points

### ISSUE THREAD

**Issue by Haoqiu-Yan** [pyg-team/pytorch_geometric#6724]
### 🐛 Describe the bug

In CiteSeer, the isolated points have been removed by transformer, and number of nodes in graph reduces to 3279.
But the dimendions of mask tensors and data.y still remain the number of 3327.

```
from torch_geometric.datasets import Planetoid
import torch_geometric.transforms as T

>>> transform = T.RemoveIsolatedNodes()
>>> dataset = Planetoid(root='~/data',
                            name="CiteSeer", split="public", transform=transform)
>>> data = dataset[0]
>>> data.num_nodes
3279
>>> len(data.x)
3279

# It seems right up to now, but when I try accessing the others,
# I got dimensions of 3327 instead of 3279 with isolated points removed.

>>> len(data.train_mask)
3327
>>> len(data.test_mask)
3327
>>> len(data.y)
3327
```

Tp remove isolated points from train_mask, test_mask and data.y, should I do something else after transform? 
I have been troubled for days, much thanks for your help :-D

### Environment

* PyG version: 2.2.0
* PyTorch version: 1.12.1
* OS: Linux
* Python version: 3.7
* CUDA/cuDNN version: 11.3
* How you installed PyTorch and PyG (`conda`, `pip`, source): conda or pip,  I don't remember very much.

**Comment by EdisonLeeeee** [comment]
This issue has been fixed in #6308. You might need to upgrade PyG from master to resolve it :)

**Comment by Haoqiu-Yan** [comment]
> This issue has been fixed in #6308. You might need to upgrade PyG from master to resolve it :)

Wow, I have installed PyG-master and it runs correctly! Thanks!!
